src/index.js

```diff
@@ -8,15 +8,15 @@ function defaultAssignEntity(normalized, key, entity) {
   normalized[key] = entity;
 }
 
-function visitObject(obj, schema, bag, options) {
+function visitObject(obj, schema, bag, options, last) {
```

Comment:
All of these function signatures are starting to get a little long. I'd like to at least see the order make more sense (optional objects last)

```
function visitObject(obj, schema, parentObj, bag, options) { ... }
```